modules/frontend/search_streaming.go

```diff
@@ -253,14 +316,19 @@ func (s *streamingSearcher) handle(r *http.Request, forwardResults func(*tempopb
 		return errors.New("request must contain a start/end date for streaming search")
 	}
 
-	progress := atomic.NewPointer[*diffSearchProgress](nil)
-	fn := func(ctx context.Context, limit, totalJobs, totalBlocks int, totalBlockBytes uint64) shardedSearchProgress {
-		p := newDiffSearchProgress(ctx, limit, totalJobs, totalBlocks, totalBlockBytes)
-		progress.Store(&p)
-		return p
+	mProgress := newMultiProgress()
+	// create diffSearchProgress for each tenant, and keep a reference to
+	// stream results back to the client.
+	progressFactoryFn := func(ctx context.Context, limit, totalJobs, totalBlocks int, totalBlockBytes uint64) shardedSearchProgress {
+		ap := atomic.NewPointer[*diffSearchProgress](nil)
```

Comment:
the atomic pointer was used to protect race conditions on the pointer access itself between the factory function and the below code. since these pointers are now protected by the multi progress object we can drop the atomic access.

Reply:
thanks, updated. that makes it simple as well :)